### ISSUE TITLE
Core-data service does not log subscription topic

### ISSUE THREAD

**Issue by M0hanrajp** [edgexfoundry/edgex-go#4843]
# 🚀 Feature Request

### Relevant Package [**REQUIRED**]

Core-data

### Description [**REQUIRED**]

My EdgeX test has the following configurations specified in the docker compose file:
* Message bus - MQTT
* Database - Redis
* Device service - Using device-virtual

The following scenario is performed:
* Bring up edgex with the mentioned configurations.
* Scenario is performed to check how data is transported from device-virtual to Redis DB ?
  * What is the interaction of message bus, core-data & any other X service. 
  * Here data being sent was either int, float, bool from device-virtual. 
  * Trace a single data (using event id or correlation id) A: device-virtual to B:Redis DB.

I have enabled (Logging level - TRACE) for all the services, **during this I found out that:**

**Why edgex-mqtt-broker has recorded that it is receiving a subscription request to the topic:
edgex/events/device/#, Where as core-data does not log the subscription event ?**
```bash
// Core-data subscribed to mqtt-broker
edgex-mqtt-broker  | 1721115601: New connection from 172.18.0.12:36492 on port 1883.
edgex-mqtt-broker  | 1721115601: New client connected from 172.18.0.12:36492 as core-data (p2, c1, k10).
edgex-mqtt-broker  | 1721115601: No will message specified.
edgex-mqtt-broker  | 1721115601: Sending CONNACK to core-data (0, 0)
edgex-mqtt-broker  | 1721115601: Received SUBSCRIBE from core-data
edgex-mqtt-broker  | 1721115601:        edgex/events/device/# (QoS 0)
edgex-mqtt-broker  | 1721115601: core-data 0 edgex/events/device/#
edgex-mqtt-broker  | 1721115601: Sending SUBACK to core-data
```

Hence I had created a discussion in order to check if my understanding is correct.
Please find Discussion#[292](https://github.com/orgs/edgexfoundry/discussions/292#discussion-6941474) for more information.

It is here it was confirmed per [Discussion#292-comment](https://github.com/orgs/edgexfoundry/discussions/292#discussioncomment-10123624) that there is no logging
event that captures information regarding core-data subscription.

### Describe the solution you'd like

I have added the logging information which will enable core-data to print the subscribed topic 
information the changes are updated in Pull request #4840 

Please note I was not aware regarding creating issue (feature request) before creating a pull 
request for enhancing the code, the current pull request is under review.

---
Thanks.

**Comment by M0hanrajp** [comment]
I have updated #4840 please help to review. 

Thanks

**Comment by cloudxxx8** [comment]
fixed in #4840
